packages/vaadin-grid/src/vaadin-grid-keyboard-navigation-mixin.js

```diff
@@ -159,22 +173,58 @@ export const KeyboardNavigationMixin = (superClass) =>
 
     /** @private */
     _ensureScrolledToIndex(index) {
-      const targetRowInDom = Array.from(this.$.items.children).filter((child) => child.index === index)[0];
+      const targetRowInDom = [...this.$.items.children].find((child) => child.index === index);
       if (!targetRowInDom) {
         this.scrollToIndex(index);
+      } else {
+        this.__scrollIntoViewport(index);
       }
     }
 
+    // TODO: A tree toggle component should not be the way to determine if the row is expandable
+    /** @private */
+    __isRowExpandable(row) {
+      const treeToggle = [...row.children].reduce(
+        (value, cell) => value || cell._content.querySelector('vaadin-grid-tree-toggle'),
+        null
+      );
```

Comment:
I think the idea behind using `reduce` here was to only run `querySelector` once?

Reply:
OK, don't have objections if it is for optimization purposes.

Reply:
Indeed, this is quite a performance-critical function since it's called on every item update for a row (to update its `aria-expanded` role). I also considered mapping but ended up with reduce because it doesn't run querySelector for the remaining row cells once it discovers the first tree toggle.

But since the optimization only works with a tree grid and this is a [temporary solution](https://github.com/vaadin/web-components/pull/2434#discussion_r703229685), I'm ok with using the more understandable version for now.

Reply:
Now I see, thanks for the detailed explanation. Actually, now I'm okay with the original version especially considering that it is called on every item update, didn't think about it before.

Reply:
...I'm about to resolve the conversation.